packages/liveblocks-react/src/factory.tsx

```diff
@@ -109,14 +136,94 @@ function makeMutationContext<
   };
 }
 
+type Options<TUserMeta extends BaseUserMeta> = {
+  /**
+   * @beta
+   *
+   * An asynchronous function that returns user info from a user ID.
+   */
+  resolveUser?: (
+    options: ResolveUserOptions
+  ) => Promise<TUserMeta["info"] | undefined>;
+
+  /**
+   * @beta
+   *
+   * An asynchronous function that returns a list of user IDs matching a string.
+   */
+  resolveMentionSuggestions?: (
+    options: ResolveMentionSuggestionsOptions
+  ) => Promise<string[]>;
+
+  /**
+   * @internal Internal endpoint
+   */
+  serverEndpoint?: string;
+};
+
+let hasWarnedIfNoResolveUser = false;
+
+function warnIfNoResolveUser(usersCache?: AsyncCache<unknown, unknown>) {
+  if (
+    !hasWarnedIfNoResolveUser &&
+    !usersCache &&
+    process.env.NODE_ENV !== "production"
+  ) {
+    console.warn(
+      "Set the resolveUser option in createRoomContext to specify user info."
+    );
+    hasWarnedIfNoResolveUser = true;
+  }
+}
+
+let hasWarnedAboutCommentsBeta = false;
+function warnIfBetaCommentsHook() {
+  if (!hasWarnedAboutCommentsBeta && process.env.NODE_ENV !== "production") {
+    console.log(
+      `%cWarning: The comments API is still in beta and may change in the future. Please report any issues you encounter at https://github.com/liveblocks/liveblocks/issues/new/choose`,
+      "color: #FFA500; font-weight: bold;"
```

Comment:
The following warning is displayed (once) when any of the following hooks are used:

- `useThreads`
- `useThreadsSuspense`
- `useCreateThread`
- `useEditThreadMetadata`
- `useCreateComment`
- `useEditComment`
- `useDeleteComment`

I have used `console.log` instead of `console.warn` because the behavior of console.warn()—including whether it shows a stack trace or not—can vary between different environments (e.g., browsers, Node.js, etc.) and is not directly configurable using standard JavaScript or APIs. This means you can't disable the stack trace in the warning if the environment decides to include one. Any thoughts on this @marcbouchenoire?

```
Warning: The Comments API is still in beta and may change in the future. Please report any issues you encounter at https://github.com/liveblocks/liveblocks/issues/new/choose 
```

Using `console.log` like described in the method (doesn't display stack trace):
<img width="1326" alt="Screenshot 2023-09-06 at 10 43 52 PM" src="https://github.com/liveblocks/liveblocks/assets/35807215/4407ed8e-6705-4213-a238-05667797a1f0">

Using `console.warn` (displays stack trace):